prboom2/src/r_draw.c

```diff
@@ -107,20 +107,24 @@ static const byte   *tempfuzzmap;
 //#define FUZZOFF (SCREENWIDTH)
 #define FUZZOFF 1
 
-static const int fuzzoffset_org[FUZZTABLE] = {
-  FUZZOFF,-FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,
-  FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,
-  FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,
-  FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF
-};
-
-static int fuzzoffset[FUZZTABLE];
-
 static int fuzzpos = 0;
 
+// Lovey01 04/29/2023: Scaled software fuzz
+static int fuzzcellsize;
+
+// NOTE: Formula is (6 + 21 - (9 - i) * (9 - i) * 5 / 19), where i is the fuzz
```

Comment:
Does this come from somewhere or did you just find something that worked?

Reply:
If you're talking about the formula, I just worked out numbers in a calculator until it matched the sequence.

Reply:
Basically what I'm asking is why and how you went about getting this formula and the values in the table. I can see that the formula and table are connected, but why this formula / why those values? Possibly it would help to spell out what `c` is in this table name or to use something else entirely. (I am also wondering if this variable name needs a comment "intensity" to explain what it is, why aren't we calling it "fuzz_intensity" and getting rid of the comment 😄)

Reply:
I actually knew this was poorly documented, basically fuzz "intensity" was how strong a fuzz cell was, or how much it darkened its contents. With the old table of offsets, since I wasn't grabbing from the pixels above or below anymore, I had to keep track of the fuzz intensity by checking whether the fuzz offset was positive or negative. Then I made a table for the intensity values and got rid of the code that handled the intensity, so now it's unknown what the "intensity" is. cmap means colormap, though that might not be the right terminology since it's an offset into tempfuzzmap. If you're wondering why I left the note here in the first place, it's for anyone planning to optimize it, because the formula could possibly the faster than the lookup.

Reply:
This documentation is more descriptive now